src/lib/spinel/spinel.h

```diff
@@ -2305,8 +2309,11 @@ enum
      *  `b`: "Is defined locally" flag. Set if this network was locally
      *       defined. Assumed to be true for set, insert and replace. Clear if
      *       the on mesh network was defined by another node.
+     *       This field is ignored for INSERT and REMOVE commands.
      *  `S`: The RLOC16 of the device that registered this on-mesh prefix entry.
      *       This value is not used and ignored when adding an on-mesh prefix.
+     *       This field is ignored for INSERT and REMOVE commands.
+     *  `C`: TLV flags extended (additional field for Thread 1.2 features).
```

Comment:
This works, but we can consider using `C: TLV flags` fields we already have here (which is used for other `SPINEL_NET_FLAGS`). I feel it may be easier (specially since new flags are `(1<<6)` and `(1<<7)` and fit in the byte. 

Thoughts?

Reply:
Hi Abtink
The 'C: TLV flags fields' is full. The 6th and 7th bits are used for storing default router preference data
('high', 'med', or 'low' status). I have added a new 'C: TLV flags fields extended' to keep additional flags separatelly.
The flags field has been added to the end of spinel structure to keep backward compatibility to previous versions.
In the INSERT command I am reading and parsing all the data until first flag field(this are mandatory data that must be present).
Additionally I am checking if "isLocal", 'rloc16' and 'extended flags' fields are present. 
These data is considered as optional.
When the fields "isLocal" and 'rloc16' are present, the variables content is ignored by INSERT command.
If the 'extended flags' data is present, it will be read and processed accordingly.


I have selected mdns flag in the extended flags bajt to be located under 7 bit and Dp flag at 6bit. 
This bits are selected to be compatible with OT kNdDnsFlag and kDpFlag constants provided in the network_data_tlvs.hpp file.

Reply:
Makes sense. Thanks @XPAWAN .